### PR TITLE
WT-2651: Coverity 1355591 resource leak

### PR DESCRIPTION
**Pull request by keithbostic**
@ddanderson, 20e0a57 is a new resource leak.

@agorrod, the rest is cleanup.

### CODE REVIEW

File: bench/wtperf/wtperf.c [hunk 1]
```diff
@@ -1858,7 +1858,7 @@ create_uris(CONFIG *cfg)
 	base_uri_len = strlen(cfg->base_uri);
 	cfg->uris = dcalloc(cfg->table_count, sizeof(char *));
 	for (i = 0; i < cfg->table_count; i++) {
-		uri = cfg->uris[i] = dcalloc(base_uri_len + 5, 1);
+		uri = cfg->uris[i] = dmalloc(base_uri_len + 5);
 		/*
```
Comment (ddanderson) [inline bench/wtperf/wtperf.c]: @keithbostic, this change to `dealloc` does not look safe. I think the memcpy code following depends on a null byte at the end of uri.

File: bench/wtperf/config.c [hunk 2]
```diff
@@ -558,7 +560,7 @@ config_opt_file(CONFIG *cfg, const char *filename)
 		goto err;
 	}
 	buf_size = (size_t)sb.st_size;
-	file_buf = dcalloc(buf_size + 2, 1);
+	file_buf = dmalloc(buf_size + 2);
```
Comment (ddanderson) [inline bench/wtperf/config.c]: @keithbostic, it looks like this could be one less byte: `dmalloc(buf_size + 1)`

File: bench/wtperf/wtperf.c [hunk 3]
```diff
@@ -1955,9 +1955,9 @@ start_all_runs(CONFIG *cfg)
 
 	home_len = strlen(cfg->home);
 	cmd_len = (home_len * 2) + 30; /* Add some slop. */
-	cmd_buf = dcalloc(cmd_len, 1);
+	cmd_buf = dmalloc(cmd_len);
 	for (i = 0; i < cfg->database_count; i++) {
-		next_cfg = dcalloc(1, sizeof(CONFIG));
+		next_cfg = dmalloc(sizeof(CONFIG));
```
Comment (ddanderson) [inline bench/wtperf/wtperf.c]: @keithbostic, is this safe? Looks like `next_cfg` as used in `config_assign` needs to be cleared, else `config_free` will access `config_head` and other fields and be very confused.

**Comment by keithbostic** [comment]
@ddanderson, you're correct, I stupidly didn't check for that. I've reverted that commit, it's not worth it.

**Comment by ddanderson** [comment]
@keithbostic, looks good now.  I'll merge when the check has completed.

**Comment by ddanderson** [comment]
retest this please

**Comment by keithbostic** [comment]
@ddanderson, @agorrod, I've reworked wtperf so it compiles on Windows.

Don, would you please do the review?

**Comment by ddanderson** [comment]
@keithbostic, LGTM.

**Pull request merged**